src/Workspaces/Core/Portable/Rename/Renamer.cs

```diff
@@ -219,5 +219,37 @@ internal static Task<LightweightRenameLocations> FindRenameLocationsAsync(Soluti
             var renameLocations = await SymbolicRenameLocations.FindLocationsInCurrentProcessAsync(symbol, solution, options, cleanupOptions, cancellationToken).ConfigureAwait(false);
             return await ConflictResolver.ResolveSymbolicLocationConflictsInCurrentProcessAsync(renameLocations, newName, nonConflictSymbolKeys, cancellationToken).ConfigureAwait(false);
         }
+
+        /// <summary>
+        /// Creates a session between the host and OOP, effectively pinning this <paramref name="solution"/> until the
+        /// supplied <paramref name="cancellationToken"/> is canceled..  By pinning the solution we ensure that all
+        /// calls to OOP for the same solution during the life of this rename session do not need to resync the
+        /// solution.  Nor do they then need to rebuild any compilations they've already built due to the solution going
+        /// away and then coming back.
+        /// </summary>
+        internal static async Task CreateKeepAliveSessionAsync(
```

Comment:
nit: Can we call this `CreateRemoteKeepAliveSessionAsync`? This is very specifically needed for OOP only and will be noop in the same process right?

Reply:
sure :)

Reply:
done.